src/core/server/saved_objects/saved_objects_service.ts

```diff
@@ -237,6 +241,11 @@ export class SavedObjectsService
     legacyTypes.forEach(type => this.typeRegistry.registerType(type));
     this.validations = setupDeps.legacyPlugins.uiExports.savedObjectValidations || {};
 
+    registerRoutes({
+      http: setupDeps.http,
+      migratorPromise: this.migrator$.pipe(first()).toPromise(),
+    });
```

Comment:
Registration is done during `setup`, the migrator is not yet created. As to use a subject/promise as workaround.